### PR TITLE
governance: Initial definition

### LINKED ISSUE #9: [CNCF] Governance
### Parts of the [CNCF onboarding issue](https://github.com/cncf/toc/issues/799) tracked by this issue

All governance related parts.

*This is the list of all bullet points from the [CNCF onboarding issue](https://github.com/cncf/toc/issues/799) that this issue will track:*

* [x] *Is your project working on written, open governance? see https://contribute.cncf.io/maintainers/governance/*
* [x] *Create maintainer list + add to aggregated https://maintainers.cncf.io/ list by submitting a PR to it*
* [x] *Provide emails for the maintainers added to https://maintainers.cncf.io/ in order to get access to the maintainers mailing list and ServiceDesk*

### PR DESCRIPTION
**Pull request by sameo**
* Roles defintion
* Steering Committee initial composition
* SC scope and Responsibilities

Fixes #9

Signed-off-by: Samuel Ortiz <samuel.e.ortiz@protonmail.com>

### CODE REVIEW

**Review by fitzthum — COMMENTED** [review]
I think this is a good starting point. I added a few comments. Realistically I think this document will have to evolve. 

Two overall points. First, there is no mention of some administrative duties like facilitating the community meetings. I think that's a pretty important aspect.

Second, I'm not sure about which aspects of the SC can happen asynchronously and which need to be in a public meeting. Is it feasible to vote on things without meeting? Do discussions between SC members that are not in meetings need to be published via some other means? These questions are probably out of scope for now.

File: governance.md
Comment (sameo) [reply governance.md:102]: I propose to have a bi-weekly (every other week) SC meeting. About reasonable times, we know very well it will always be unreasonable for at least one time zone. I think the timing  should be left for the SC members to decide and agree upon and as such it may change over time.
Also, yes those meetings should be public and recorded, I think.

File: governance.md
Comment (sameo) [reply governance.md:105]: Do we need to define what we are voting on? This is defining how a vote should be accepted, if the SC decides to vote on something. IMHO, voting should be a last resort kind of thing. We could define a process for any member to call for a vote, but that's pretty heavy at that point.

And yes, SC members should start right away working on amending that document with anything they believe should be added (Although I'm hoping discussions on that PR should lead us to something all 10 members largely agree on), and in particular define the SC renewal/extension process.

Comment (jiangliu) [reply governance.md:105]: Should we have an odd number of members?

File: governance.md
Comment (sameo) [reply governance.md]: Thanks, fixed now.

File: governance.md
Comment (sameo) [reply governance.md]: Good points. Fixed as well.

File: governance.md
Comment (larrydewey) [reply governance.md:8]: Agreed. A re-wording of this would be beneficial. We obviously would want to keep things like CVE's, etc, private until appropriate actions are taken/tested, approved, and released. That being said, I do like highlighting that we are as transparent as possible. Would it be beneficial to add in a small snippet about licensing here, as well?

File: governance.md
Comment (larrydewey) [inline governance.md:28]: Just a knit, maybe re-word to:

**Adding or enhancing** documentation, tests...

File: governance.md
Comment (larrydewey) [inline governance.md:36]: Just a knit:

*Maintiner~~s~~*

File: governance.md
Comment (larrydewey) [reply governance.md:106]: Holding a bi-weekly meeting makes sense to me, as well. I would be open to possibly discussing an alternating meeting schedule, as well. I don't want to over complicate or make the meeting times too confusing, but it might be nice for those for which the meeting is less convenient?

File: governance.md
Comment (larrydewey) [reply governance.md:109]: > Should we have an odd number of members?

This brings ups an important point. If we have a split decision, having an even number of committee members could potentially stall during the vote.

**Review by larrydewey — COMMENTED** [review]
This looks like a really good start for our initial definition. Thanks @sameo for putting this all together! I mentioned it in #9, but as the project become more popular, will the requirements for consideration for entry into the SC also scale upward? I am curious about this, as depending on activity, this could be monopolizing in either direction, as code or documentation specific contributions by count could potentially leave out weights of importance or quality.

File: governance.md
Comment (fitzthum) [reply governance.md:109]: Note that the quorum for voting isn't the full committee. Regardless of the size of the committee there is no guarantee that an odd number of people would show up to vote.

Comment (larrydewey) [reply governance.md:109]: > Regardless of the size of the committee there is no guarantee that an odd number of people would show up to vote.

That is a valid point. It is likely that most scenarios at least one member will likely be missing. That being said, I think it is probably a good idea to have prepared for it before hand.

> I guess it's not ideal if everyone were to show up and the committee ends up split, but this doesn't keep me up at night.
 
It doesn't keep me up, either, it was more of an observation. I just wouldn't want a split committee preventing the progress of the project. Given the character of the members, and the fact that it hasn't really been an issue up to this point, I would venture to guess it won't be a major issue in the near future.

File: governance.md
Comment (jiangliu) [reply governance.md:8]: Absolutely we need commercial friendly license models. But should it be defined by the TOC team?

File: governance.md
Comment (sameo) [reply governance.md:109]: I agree with @fitzthum here, basically the SC size is vaguely related to the number of votes.

Moreover, and given the scope of the SC and at least its initial composition, if we end up in a situation where it's split on a given decision, it fundamentally means there is a community disagreement. Personally, I am not in favor of resolving such contention through e.g. a 6 vs 5 voting result (Assuming a 11 SC members). The fundamental role of the SC in that case would be more about finding a way out of it that does not potentially leave almost half of the community unhappy with a decision.

And yes, I dont expect this is a situation that we'll face often, if ever. So, in the spirit of keeping things simple, I'd prefer to not be too formal with those details.

File: governance.md
Comment (sameo) [reply governance.md:8]: The licensing is already specified in the `CONTRIBUTING` document, but I'm adding the fact that the project is licensed under the Apache 2 license in the `Principles` section.

File: governance.md
Comment (sameo) [reply governance.md:28]: Fixed, thanks.

File: governance.md
Comment (sameo) [reply governance.md:36]: Fixed, thanks.

File: governance.md
Comment (sameo) [reply governance.md:9]: I'll also reword the `Transparent` part.

**Review by fitzthum — APPROVED** [review]

File: governance.md
Comment (dcmiddle) [reply governance.md:115]: I like the description of consensus driven decision making in the Rust community. 
https://github.com/rust-lang/rfcs/blob/master/text/1068-rust-governance.md#consensus
It would be good to add similar language here. The style of our document is much more succinct, but perhaps we can link to broader text like Rust's.

Comment (sameo) [reply governance.md:129]: Thanks for the suggestion @dcmiddle. I added a small section about decision-making.

**Comment by sameo** [comment]
@fitzthum @jiangliu @dcmiddle @larrydewey I think I addressed most (all?) comments on this PR. The last revision includes a decision-making section and a slightly more realistic wording on our transparency.

Please let me know if you think this is a good enough starting point, or if you have any other concerns/comments you'd like to see addressed.

File: governance.md
Comment (magowan) [inline governance.md:52]: Maybe it is overkill here but perhaps highlighting that the Steering Committee can also be a route to becoming a maintainer? What if current maintainers simply don't elevate new Contributors or as can happen we end up with no active maintainers to "elevate" anyone new?

File: governance.md
Comment (magowan) [inline governance.md:81]: Maybe another example here is actively looking to ensure there are always enough active maintainers for each project?

File: governance.md
Comment (magowan) [reply governance.md:84]: I like this suggestion and think it is important to demonstrate the importance of inclusion with active element.
I can't vouch for a particular course as those I have done are internal but I would be happy to make myself familiar with the material if people see @dcmiddle suggested material as our reference here?
(Merely an issue of not enough hours to read all material if others get suggested)

File: governance.md
Comment (magowan) [inline governance.md:82]: Perhps add inclusive and transparent here? To emphasise these key attributes?
* Building and growing a transparent and inclusive Confidential Containers community

File: governance.md
Comment (magowan) [inline governance.md:113]: nit - unadressed -> unaddressed

File: governance.md
Comment (magowan) [reply governance.md:126]: I am open to alternating schedule - though I appreciate that being in Europe I am perhaps least affected by possible meeting times.

File: governance.md
Comment (magowan) [reply governance.md:129]: I like the wording at present as a good starting point and I think any "vote" that is close should lead to further discussion - and if the steering group itself cannot resolve it then I wouldn't see an issue encouraging and inviting  all in community to discuss and help reach a decision? This will hopefully never be needed but can be our last resort option?
Ultimately there is no last resort great option here , we can't control the numbers attending any meeting and if a vote is close unless we address the major concerns any decision will be less sticky and no doubt come back again in near future.

**Review by magowan — COMMENTED** [review]
I like the content and agree with the principle get a starting point up there and lets work together to refine this through discussion to a more complete document/processes

**Review by jiazhang0 — APPROVED** [review]

File: governance.md
Comment (dcmiddle) [reply governance.md:84]: The advantage to this course is that it focuses on issues specific to open source. At the same time, I tried to word it such that people are not explicitly required to take it but that this leadership role requires the contained subject matter whether you get it through this course or similar training or experience.

**Review by jiangliu — APPROVED** [review]

**Review by larrydewey — APPROVED** [review]

File: governance.md
Comment (sameo) [reply governance.md]: Done, thanks.

File: governance.md
Comment (sameo) [reply governance.md]: Thanks, fixed.

File: governance.md
Comment (sameo) [reply governance.md:84]: @dcmiddle @magowan I added Dan's suggestion, thanks.

File: governance.md
Comment (sameo) [reply governance.md]: Looks good. I change that part with your input, @magowan

**Comment by sameo** [comment]
@magowan I think I addressed most of your comments. Could you please have another look?

**Review by dcmiddle — APPROVED** [review]

File: governance.md
Comment (vbatts) [inline governance.md:48]: :+1: simple enough. Maybe it's worth saying the CODEOWNERS at the top-level of the repo, for the `*` glob?

Because potential drawback is in the future having nested CODEOWNERS to isolate folks for particular subsystem subject-matter-expertise, or different glob patterns than just `*`

**Review by vbatts — COMMENTED** [review]
overall looks like a good start.

on nit: maybe a format of one sentence per line, with paragraphs separated by the extra newline. The markdown rendering still looks the same, but it makes the diffs more sentence specific. Not a huge deal, but has made some doc review simpler.

**Review by magowan — APPROVED** [review]
Good with me, thanks

File: governance.md
Comment (sameo) [reply governance.md]: Done, thanks.

**Comment by sameo** [comment]
> overall looks like a good start.
> 
> on nit: maybe a format of one sentence per line, with paragraphs separated by the extra newline. The markdown rendering still looks the same, but it makes the diffs more sentence specific. Not a huge deal, but has made some doc review simpler.

Good suggestion, I followed it now.

**Pull request merged**
